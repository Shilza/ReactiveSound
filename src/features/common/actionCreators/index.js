import {
    FETCHED_TRACK,
    PAUSE_TRACK,
    PLAY_TRACK,
    SEEK_TO,
    SET_CURRENT_TIME, SET_PLAYER,
    SET_PLAYER_INTERVAL,
    FETCHED_NEXT,
    FETCHED_PREVIOUS, FETCHED_WAVEFORM_TRACKS, REQUESTED_WAVEFORM_SUCCEED
} from "../actionTypes";

export const fetchTrack = payload => ({type: FETCHED_TRACK, payload});

export const setPlayer = payload => ({type: SET_PLAYER, payload});

export const setPlayerInterval = payload => ({type: SET_PLAYER_INTERVAL, payload});

export const setCurrentTime = payload => ({type: SET_CURRENT_TIME, payload});

export const playTrack = () => ({type: PLAY_TRACK});

export const pauseTrack = () => ({type: PAUSE_TRACK});

export const seekTo = payload => ({type: SEEK_TO, payload});

export const fetchNext = payload => ({type: FETCHED_NEXT, payload});

export const fetchPrevious = payload => ({type: FETCHED_PREVIOUS, payload});


export const fetchWaveform = payload => ({type: FETCHED_WAVEFORM_TRACKS, payload});

export const requestWaveformSuccess = payload => ({type: REQUESTED_WAVEFORM_SUCCEED, payload});