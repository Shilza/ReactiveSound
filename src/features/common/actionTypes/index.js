
export const FETCHED_TRACK = 'FETCHED_TRACK';
export const FETCHED_NEXT = 'FETCHED_NEXT';
export const FETCHED_PREVIOUS = 'FETCHED_PREVIOUS';
export const SET_PLAYER = 'SET_PLAYER';
export const PLAY_TRACK = 'PLAY_TRACK';
export const PAUSE_TRACK = 'PAUSE_TRACK';
export const SET_PLAYER_INTERVAL = 'SET_PLAYER_INTERVAL';
export const SET_CURRENT_TIME = 'SET_CURRENT_TIME';
export const SEEK_TO = 'SEEK_TO';

export const FETCHED_WAVEFORM_TRACKS = 'FETCHED_WAVEFORM_TRACKS';
export const REQUESTED_WAVEFORM_SUCCEED = 'REQUESTED_WAVEFORM_SUCCEED';