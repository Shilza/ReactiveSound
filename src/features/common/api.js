import SC from "soundcloud";

const LIMIT = 22;
const LINKED_PARTITIONING = 1;

const get = (url, body) => SC.get(url, {
    limit: LIMIT,
    linked_partitioning: LINKED_PARTITIONING,
    ...body
}).then(tracks => tracks);

const getTrackById = track =>
    SC.stream(`/tracks/${track.id}`).then(sound =>
        ({
            player: sound,
            currentTrack: track
        })
    );

const getFavorites = (id = 185676792) => get(`/users/${id}/favorites`);

const getUsersTracksById = id => get(`/users/${id}/tracks`);

const getTracks = query => get('/tracks', {q: query});

const getUserById = id => get(`/users/${id}`);

export const tracksApi = {
    getTrackById,
    getFavorites,
    getTracks,
    getUsersTracksById,
    getUserById
};