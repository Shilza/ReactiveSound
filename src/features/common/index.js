
export { CommonContent } from './templates';
export { NoMatch } from './pages/NoMatch';