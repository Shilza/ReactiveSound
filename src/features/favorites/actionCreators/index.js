import {
    FETCHED_FAVORITE_TRACKS,
    REQUESTED_FAVORITE_TRACKS,
    REQUESTED_FAVORITE_TRACKS_FAILED,
    REQUESTED_FAVORITE_TRACKS_SUCCEED,
} from "../actionTypes";

export const requestFavoriteTracks = () => ({ type: REQUESTED_FAVORITE_TRACKS });

export const requestFavoriteTracksSuccess = data => ({ type: REQUESTED_FAVORITE_TRACKS_SUCCEED, payload: data });

export const requestFavoriteTracksError = () => ({ type: REQUESTED_FAVORITE_TRACKS_FAILED });

export const fetchFavoriteTracks = () => ({ type: FETCHED_FAVORITE_TRACKS });

