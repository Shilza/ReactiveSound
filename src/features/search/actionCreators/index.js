import {
    FETCHED_SEARCH_TRACKS, FETCHED_WAVEFORM_TRACKS,
    REQUESTED_SEARCH_TRACKS,
    REQUESTED_SEARCH_TRACKS_FAILED,
    REQUESTED_SEARCH_TRACKS_SUCCEED, REQUESTED_WAVEFORM_SUCCEED
} from "../actionTypes";

export const requestSearchTracks = () => ({ type: REQUESTED_SEARCH_TRACKS });

export const requestSearchTracksSuccess = payload => ({ type: REQUESTED_SEARCH_TRACKS_SUCCEED, payload });

export const requestSearchTracksError = () => ({ type: REQUESTED_SEARCH_TRACKS_FAILED });

export const fetchSearchTracks = payload => ({ type: FETCHED_SEARCH_TRACKS, payload });


export const fetchWaveform = payload => ({ type: FETCHED_WAVEFORM_TRACKS, payload });

export const requestWaveformSuccess = payload => ({ type: REQUESTED_WAVEFORM_SUCCEED, payload });