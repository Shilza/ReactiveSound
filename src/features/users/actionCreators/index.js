import {
    FETCHED_USER, FETCHED_USERS_LIKED_TRACKS, FETCHED_USERS_LIKED_TRACKS_BY_PAGE,
    FETCHED_USERS_TRACKS, FETCHED_USERS_TRACKS_BY_PAGE,
    REQUESTED_USER,
    REQUESTED_USER_FAILED,
    REQUESTED_USER_SUCCEED,
    REQUESTED_USERS_LIKED_TRACKS, REQUESTED_USERS_LIKED_TRACKS_FAILED,
    REQUESTED_USERS_LIKED_TRACKS_SUCCEED,
    REQUESTED_USERS_TRACKS,
    REQUESTED_USERS_TRACKS_FAILED,
    REQUESTED_USERS_TRACKS_SUCCEED, RESET_LIKED_TRACKS, RESET_USERS_TRACKS
} from "../actionTypes";

export const requestUsersTracks = () => ({ type: REQUESTED_USERS_TRACKS });

export const fetchUsersTracksByPage = () => ({ type: FETCHED_USERS_TRACKS_BY_PAGE });

export const requestUsersTracksSuccess = payload => ({ type: REQUESTED_USERS_TRACKS_SUCCEED, payload });

export const requestUsersTracksError = () => ({ type: REQUESTED_USERS_TRACKS_FAILED });

export const fetchUsersTracks = payload => ({ type: FETCHED_USERS_TRACKS, payload });

export const resetUsersTracks = ()  => ({ type: RESET_USERS_TRACKS });


export const requestUsersLikedTracks = () => ({ type: REQUESTED_USERS_LIKED_TRACKS });

export const fetchLikedTracksByPage = () => ({ type: FETCHED_USERS_LIKED_TRACKS_BY_PAGE });

export const requestUsersLikedTracksSuccess = payload => ({ type: REQUESTED_USERS_LIKED_TRACKS_SUCCEED, payload });

export const requestUsersLikedTracksError = () => ({ type: REQUESTED_USERS_LIKED_TRACKS_FAILED });

export const fetchUsersLikedTracks = payload => ({ type: FETCHED_USERS_LIKED_TRACKS, payload });

export const resetLikedTracks = ()  => ({ type: RESET_LIKED_TRACKS });


export const requestUser = () => ({ type: REQUESTED_USER });

export const requestUserSuccess = payload => ({ type: REQUESTED_USER_SUCCEED, payload });

export const requestUserError = () => ({ type: REQUESTED_USER_FAILED });

export const fetchUser = payload => ({ type: FETCHED_USER, payload });