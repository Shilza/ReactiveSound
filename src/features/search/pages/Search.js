import React, {useEffect} from "react";
import {CommonContent} from "../../common/templates";
import {SubHeader} from "../../favorites/organisms";
import {connect} from "react-redux";
import {fetchSearchTracks} from "../actionCreators";
import {Tracks} from "../../favorites/templates/Tracks";

const Search = ({match: {params}, dispatch, loading, tracks}) => {

    useEffect(() => {
        dispatch(fetchSearchTracks(params.query))
    }, [params, dispatch]);

    return (
        <CommonContent>
            <SubHeader location={params.query}/>
            <Tracks tracks={tracks} loading={loading}/>
        </CommonContent>
    );
};

export default connect(state => ({
    error: state.search.search.error,
    loading: state.search.search.loading,
    tracks: state.search.search.tracks
}))(Search);