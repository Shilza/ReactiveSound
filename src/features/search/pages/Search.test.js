import React from 'react';
import {render} from 'react-testing-library';
import {BrowserRouter as Router} from "react-router-dom";
import Search from "./Search";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {fetchSearchTracks} from "../actionCreators";
import {tracksApi} from "../../common/api";

jest.mock('../../common/api');

const resolvedValue = [
    {
        id: 500
    }
];
tracksApi.getTracks.mockResolvedValue(resolvedValue);

describe('Test Search page', () => {

    const mockStore = configureStore();
    const store = mockStore({
        search: {
            error: false,
            loading: false,
            tracks: [],
            nextPage: undefined
        }
    });

    const query = 'Artist';
    const {container} = render(
        <Provider store={store}>
            <Router>
                <Search match={{params: {query}}}/>
            </Router>
        </Provider>
    );

    it('should renders correctly', () => {
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should load tracks', () => {
        expect(store.getActions()[0]).toEqual(fetchSearchTracks('Artist'));
    });
});