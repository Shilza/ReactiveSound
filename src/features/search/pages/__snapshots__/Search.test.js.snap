// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Search page should renders correctly 1`] = `
<header
  class="container"
>
  <div
    class="header"
  >
    <h1
      class="title"
    >
      <a
        class="link"
        href="/"
      >
        Reactive Sound
      </a>
    </h1>
    <nav
      class="container"
    >
      <ul
        class="iconsList"
      >
        <li>
          <button
            class="button"
            data-testid="searchButton"
            id="searchButton"
          >
            <svg
              fill="#7f8081"
              focusable="false"
              height="1.1em"
              id="searchIcon"
              role="img"
              viewBox="0 0 500 500"
              width="1.1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M447.05,428l-109.6-109.6c29.4-33.8,47.2-77.9,47.2-126.1C384.65,86.2,298.35,0,192.35,0C86.25,0,0.05,86.3,0.05,192.3
 s86.3,192.3,192.3,192.3c48.2,0,92.3-17.8,126.1-47.2L428.05,447c2.6,2.6,6.1,4,9.5,4s6.9-1.3,9.5-4
 C452.25,441.8,452.25,433.2,447.05,428z
 M26.95,192.3c0-91.2,74.2-165.3,165.3-165.3c91.2,0,165.3,74.2,165.3,165.3
 s-74.1,165.4-165.3,165.4C101.15,357.7,26.95,283.5,26.95,192.3z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            class="button"
          >
            <a
              href="http://soundcloud.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                fill="#7f8081"
                focusable="false"
                height="1.1em"
                role="img"
                viewBox="0 0 100 100"
                width="1.1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.361,67.281L5,60.279l-1.639-7.238c-0.045-0.215-0.457-0.379-0.959-0.379
 c-0.508,0-0.92,0.164-0.959,0.381L0,60.279l1.443,7.002c0.039,0.217,0.451,0.381,0.959,0.381
 C2.904,67.662,3.316,67.498,3.361,67.281z
 M13.611,71.168L15,60.348l-1.389-16.74c-0.041-0.426-0.561-0.76-1.191-0.76
 c-0.635,0-1.156,0.334-1.188,0.76L10,60.348l1.232,10.82c0.031,0.422,0.553,0.756,1.188,0.756
 C13.051,71.924,13.57,71.594,13.611,71.168z
 M23.84,70.99L25,60.354l-1.16-22.287c-0.031-0.523-0.648-0.934-1.404-0.934
 c-0.762,0-1.379,0.41-1.406,0.934L20,60.354l1.029,10.642c0.027,0.519,0.645,0.928,1.406,0.928
 C23.191,71.922,23.809,71.514,23.84,70.99z
 M34.049,70.832L35,60.355l-0.951-22.449c-0.023-0.621-0.727-1.107-1.6-1.107
 c-0.879,0-1.582,0.486-1.604,1.107L30,60.355l0.85,10.475c0.018,0.615,0.721,1.102,1.6,1.102
 C33.322,71.932,34.025,71.453,34.049,70.832z
 M44.248,70.699L45,60.359l-0.752-25.74c-0.016-0.707-0.818-1.281-1.787-1.281
 c-0.971,0-1.771,0.576-1.787,1.281L40,60.352c0,0.017,0.674,10.349,0.674,10.349c0.016,0.698,0.816,1.272,1.787,1.272
 C43.43,71.973,44.232,71.406,44.248,70.699z
 M51.391,71.98C51.424,71.982,86.883,72,87.114,72C94.232,72,100,66.42,100,59.537
 c0-6.885-5.768-12.465-12.887-12.465c-1.766,0-3.449,0.348-4.984,0.969C81.104,36.811,71.363,28,59.484,28
 c-2.906,0-5.742,0.553-8.244,1.488c-0.972,0.366-1.232,0.739-1.24,1.467v39.553C50.01,71.27,50.621,71.906,51.391,71.98z"
                />
              </svg>
            </a>
          </button>
        </li>
        <li>
          <button
            class="button"
          >
            <a
              href="https://github.com/Shilza/ReactiveSound"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                fill="#7f8081"
                focusable="false"
                height="1.1em"
                role="img"
                viewBox="0 0 430 430"
                width="1.1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M409.132,114.573c-19.608-33.596-46.205-60.194-79.798-79.8C295.736,15.166,259.057,5.365,219.271,5.365
 c-39.781,0-76.472,9.804-110.063,29.408c-33.596,19.605-60.192,46.204-79.8,79.8C9.803,148.168,0,184.854,0,224.63
 c0,47.78,13.94,90.745,41.827,128.906c27.884,38.164,63.906,64.572,108.063,79.227c5.14,0.954,8.945,0.283,11.419-1.996
 c2.475-2.282,3.711-5.14,3.711-8.562c0-0.571-0.049-5.708-0.144-15.417c-0.098-9.709-0.144-18.179-0.144-25.406l-6.567,1.136
 c-4.187,0.767-9.469,1.092-15.846,1c-6.374-0.089-12.991-0.757-19.842-1.999c-6.854-1.231-13.229-4.086-19.13-8.559
 c-5.898-4.473-10.085-10.328-12.56-17.556l-2.855-6.57c-1.903-4.374-4.899-9.233-8.992-14.559
 c-4.093-5.331-8.232-8.945-12.419-10.848l-1.999-1.431c-1.332-0.951-2.568-2.098-3.711-3.429c-1.142-1.331-1.997-2.663-2.568-3.997
 c-0.572-1.335-0.098-2.43,1.427-3.289c1.525-0.859,4.281-1.276,8.28-1.276l5.708,0.853c3.807,0.763,8.516,3.042,14.133,6.851
 c5.614,3.806,10.229,8.754,13.846,14.842c4.38,7.806,9.657,13.754,15.846,17.847c6.184,4.093,12.419,6.136,18.699,6.136
 c6.28,0,11.704-0.476,16.274-1.423c4.565-0.952,8.848-2.383,12.847-4.285c1.713-12.758,6.377-22.559,13.988-29.41
 c-10.848-1.14-20.601-2.857-29.264-5.14c-8.658-2.286-17.605-5.996-26.835-11.14c-9.235-5.137-16.896-11.516-22.985-19.126
 c-6.09-7.614-11.088-17.61-14.987-29.979c-3.901-12.374-5.852-26.648-5.852-42.826c0-23.035,7.52-42.637,22.557-58.817
 c-7.044-17.318-6.379-36.732,1.997-58.24c5.52-1.715,13.706-0.428,24.554,3.853c10.85,4.283,18.794,7.952,23.84,10.994
 c5.046,3.041,9.089,5.618,12.135,7.708c17.705-4.947,35.976-7.421,54.818-7.421s37.117,2.474,54.823,7.421l10.849-6.849
 c7.419-4.57,16.18-8.758,26.262-12.565c10.088-3.805,17.802-4.853,23.134-3.138c8.562,21.509,9.325,40.922,2.279,58.24
 c15.036,16.18,22.559,35.787,22.559,58.817c0,16.178-1.958,30.497-5.853,42.966c-3.9,12.471-8.941,22.457-15.125,29.979
 c-6.191,7.521-13.901,13.85-23.131,18.986c-9.232,5.14-18.182,8.85-26.84,11.136c-8.662,2.286-18.415,4.004-29.263,5.146
 c9.894,8.562,14.842,22.077,14.842,40.539v60.237c0,3.422,1.19,6.279,3.572,8.562c2.379,2.279,6.136,2.95,11.276,1.995
 c44.163-14.653,80.185-41.062,108.068-79.226c27.88-38.161,41.825-81.126,41.825-128.906
 C438.536,184.851,428.728,148.168,409.132,114.573z"
                />
              </svg>
            </a>
          </button>
        </li>
      </ul>
    </nav>
  </div>
  <div
    class="searchBarLeave"
    data-testid="searchBar"
  >
    <form
      class="barContainer"
      data-testid="searchForm"
    >
      <input
        class="searchInput"
        data-testid="searchInput"
        maxlength="60"
        placeholder="Search Tracks"
        type="text"
      />
    </form>
  </div>
</header>
`;
